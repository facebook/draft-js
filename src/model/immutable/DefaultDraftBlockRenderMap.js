/**
 * Copyright (c) 2013-present, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 *
 * @providesModule DefaultDraftBlockRenderMap
 * @flow
 */

'use strict';

const {Map} = require('immutable');
const React = require('React');

const cx = require('cx');

const UL_WRAP = <ul className={cx('public/DraftStyleDefault/ul')} />;
const OL_WRAP = <ol className={cx('public/DraftStyleDefault/ol')} />;
const PRE_WRAP = <pre className={cx('public/DraftStyleDefault/pre')} />;

module.exports = Map({
  'header-one': {
    element: 'h1',
    nestingEnabled: false
  },
  'header-two': {
    element: 'h2',
    nestingEnabled: false
  },
  'header-three': {
    element: 'h3',
    nestingEnabled: false
  },
  'header-four': {
    element: 'h4',
    nestingEnabled: false
  },
  'header-five': {
    element: 'h5',
    nestingEnabled: false
  },
  'header-six': {
    element: 'h6',
    nestingEnabled: false
  },
  'unordered-list-item': {
    element: 'li',
    wrapper: UL_WRAP,
    nestingEnabled: false
  },
  'ordered-list-item': {
    element: 'li',
    wrapper: OL_WRAP,
    nestingEnabled: false
  },
  'blockquote': {
    element: 'blockquote',
    nestingEnabled: false
  },
  'atomic': {
    element: 'figure',
    nestingEnabled: false
  },
  'code-block': {
    element: 'pre',
    wrapper: PRE_WRAP,
    nestingEnabled: false
  },
  'unstyled': {
    element: 'div',
    nestingEnabled: false
  },
});
