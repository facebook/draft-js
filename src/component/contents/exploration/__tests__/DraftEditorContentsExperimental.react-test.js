/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @emails oncall+draft_js
 * @format
 */

'use strict';

import ContentBlockNode from 'ContentBlockNode';
import ContentState from 'ContentState';
import DefaultDraftBlockRenderMap from 'DefaultDraftBlockRenderMap';
import DraftEditorContentsExperimental from 'DraftEditorContentsExperimental.react';
import EditorState from 'EditorState';
import * as React from 'React';
import SelectionState from 'SelectionState';

import {transformSnapshotProps} from '_DraftTestHelper';
import Immutable from 'immutable';
import ReactTestRenderer from 'react-test-renderer';

const DraftEditorContents = DraftEditorContentsExperimental;

const {List} = Immutable;

const selectionState = new SelectionState({
  anchorKey: 'A',
  anchorOffset: 0,
  focusKey: 'A',
  focusOffset: 0,
  isBackward: false,
  hasFocus: true,
});

const contentState = ContentState.createFromBlockArray([
  new ContentBlockNode({
    key: 'A',
    text: 'Alpha',
    type: 'blockquote',
    children: List(),
  }),
]);

const PROPS = {
  blockRenderMap: DefaultDraftBlockRenderMap,
  blockRendererFn: block => null,
  blockStyleFn: block => '',
  contentState,
  customStyleFn: (style, block) => null,
  editorKey: 'editor',
  editorState: EditorState.createWithContent(contentState),
  selection: selectionState,
};

const assertDraftEditorContentsRendering = props => {
  const childProps = {
    ...props,
    editorState: EditorState.createWithContent(props.contentState),
  };

  const blockNode = ReactTestRenderer.create(
    <DraftEditorContents {...childProps} />,
  );

  expect(transformSnapshotProps(blockNode.toJSON())).toMatchSnapshot();
};

test('renders ContentBlockNode', () => {
  assertDraftEditorContentsRendering(PROPS);
});

test('renders ContentBlockNodes', () => {
  const contentState = ContentState.createFromBlockArray([
    new ContentBlockNode({
      key: 'A',
      text: '',
      type: 'blockquote',
      children: List(['B']),
      nextSibling: 'D',
    }),
    new ContentBlockNode({
      parent: 'A',
      key: 'B',
      text: 'fist list item',
      type: 'ordered-list-item',
      children: List(['C']),
    }),
    new ContentBlockNode({
      parent: 'B',
      key: 'C',
      text: 'header inside list',
      type: 'header-one',
    }),
    new ContentBlockNode({
      key: 'D',
      prevSibling: 'A',
      text: 'header two',
      type: 'header-two',
    }),
  ]);

  assertDraftEditorContentsRendering({
    ...PROPS,
    contentState,
  });
});

test('renders ContentBlockNodes with root blocks that have wrapperTemplate', () => {
  const contentState = ContentState.createFromBlockArray([
    new ContentBlockNode({
      key: 'A',
      text: 'list one',
      type: 'ordered-list-item',
      nextSibling: 'B',
    }),
    new ContentBlockNode({
      key: 'B',
      text: 'list two',
      type: 'ordered-list-item',
      prevSibling: 'A',
    }),
  ]);

  assertDraftEditorContentsRendering({
    ...PROPS,
    contentState,
  });
});
