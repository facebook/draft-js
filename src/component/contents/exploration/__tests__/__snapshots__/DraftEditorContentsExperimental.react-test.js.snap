// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ContentBlockNodes 1`] = `
<div
  data-contents="true"
>
  <blockquote
    data-block={true}
    data-editor="editor"
    data-offset-key="A-0-0"
  >
    <div
      data-offset-key="A-0-0"
    >
      <ol
        data-offset-key="B-0-0"
      >
        <li
          data-block={true}
          data-editor="editor"
          data-offset-key="B-0-0"
        >
          <div
            data-offset-key="B-0-0"
          >
            <h1
              data-block={true}
              data-editor="editor"
              data-offset-key="C-0-0"
            >
              <div
                data-offset-key="C-0-0"
              >
                <span
                  data-offset-key="C-0-0"
                  style={Object {}}
                >
                  <span
                    data-text="true"
                  >
                    header inside list
                  </span>
                </span>
              </div>
            </h1>
          </div>
        </li>
      </ol>
    </div>
  </blockquote>
  <h2
    data-block={true}
    data-editor="editor"
    data-offset-key="D-0-0"
  >
    <div
      data-offset-key="D-0-0"
    >
      <span
        data-offset-key="D-0-0"
        style={Object {}}
      >
        <span
          data-text="true"
        >
          header two
        </span>
      </span>
    </div>
  </h2>
</div>
`;

exports[`renders ContentBlockNodes with root blocks that have wrapperTemplate 1`] = `
<div
  data-contents="true"
>
  <ol
    data-offset-key="A-0-0"
  >
    <li
      data-block={true}
      data-editor="editor"
      data-offset-key="A-0-0"
    >
      <div
        data-offset-key="A-0-0"
      >
        <span
          data-offset-key="A-0-0"
          style={Object {}}
        >
          <span
            data-text="true"
          >
            list one
          </span>
        </span>
      </div>
    </li>
    <li
      data-block={true}
      data-editor="editor"
      data-offset-key="B-0-0"
    >
      <div
        data-offset-key="B-0-0"
      >
        <span
          data-offset-key="B-0-0"
          style={Object {}}
        >
          <span
            data-text="true"
          >
            list two
          </span>
        </span>
      </div>
    </li>
  </ol>
</div>
`;

exports[`renders ContentBlocks 1`] = `
<div
  data-contents="true"
>
  <blockquote
    data-block={true}
    data-editor="editor"
    data-offset-key="A-0-0"
  >
    <div
      data-offset-key="A-0-0"
    >
      <span
        data-offset-key="A-0-0"
        style={Object {}}
      >
        <span
          data-text="true"
        >
          Alpha
        </span>
      </span>
    </div>
  </blockquote>
  <h1
    data-block={true}
    data-editor="editor"
    data-offset-key="B-0-0"
  >
    <div
      data-offset-key="B-0-0"
    >
      <span
        data-offset-key="B-0-0"
        style={Object {}}
      >
        <span
          data-text="true"
        >
          Beta
        </span>
      </span>
    </div>
  </h1>
  <div
    data-block={true}
    data-editor="editor"
    data-offset-key="C-0-0"
  >
    <div
      data-offset-key="C-0-0"
    >
      <span
        data-offset-key="C-0-0"
        style={Object {}}
      >
        <span
          data-text="true"
        >
          Charlie
        </span>
      </span>
    </div>
  </div>
</div>
`;
